StorageUtility/Windows/Device.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "StorageUtility.h"
+
+namespace vtStor
+{
+
+cDevice::cDevice(PSP_DEVINFO_DATA DevInfoData, PSP_DEVICE_INTERFACE_DATA DevInterfaceData, PSP_INTERFACE_DEVICE_DETAIL_DATA DevDetailData)
+{
+    m_Data.clear();
+    m_Data.insert({ eDeviceDataType::PSP_DEVINFO_DATA, DevInfoData });
+    m_Data.insert({ eDeviceDataType::PSP_DEVICE_INTERFACE_DATA, DevInterfaceData });
+    m_Data.insert({ eDeviceDataType::PSP_INTERFACE_DEVICE_DETAIL_DATA, DevDetailData });
```

Comment:
1. Remove m_Data, because it's not necessary.
2. Create deep copies of the device data structures.